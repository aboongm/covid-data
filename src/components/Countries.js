import React from 'react';
import { FaRegArrowAltCircleRight } from 'react-icons/fa';
import { useNavigate } from 'react-router-dom';
import PropTypes from 'prop-types';
import { motion } from 'framer-motion';

const Countries = (props) => {
  const {
    data: { name, today_confirmed: TotalConfirmed },
  } = props;
  const navigate = useNavigate();
  return (
    <motion.div
      className=" col-6"
      initial={{ x: '-100vw' }}
      animate={{ x: 1 }}
      transition={{ delay: 0.2, duration: 0.8 }}
    >
      <div className="countries">
        <button
          type="button"
          className="navigate"
          onClick={() => navigate(`/${name}`)}
        >
          {name}
        </button>
        <p className="text-white text2">{TotalConfirmed}</p>
        {' '}
        <FaRegArrowAltCircleRight
          className="arrow-right"
          onClick={() => navigate(`/${name}`)}
        />
      </div>
    </motion.div>
  );
};

Countries.propTypes = {
  data: PropTypes.shape({
    today_confirmed: PropTypes.number,
    name: PropTypes.string,
  }).isRequired,
};

export default Countries;
