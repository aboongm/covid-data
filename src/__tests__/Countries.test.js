import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import Countries from '../components/Countries';

describe('Countries tests', () => {
  test('should render', () => {
    const data = {
      name: 'UK',
      TotalConfirmed: 1000,
    };
    const countries = render(
      <BrowserRouter>
        <Countries data={data} />
      </BrowserRouter>,
    );

    expect(countries).toMatchSnapshot();
  });
});
