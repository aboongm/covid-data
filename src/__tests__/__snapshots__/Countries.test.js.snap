// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Countries tests should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class=" col-6"
        style="transform: none;"
      >
        <div
          class="countries"
        >
          <button
            class="navigate"
            type="button"
          >
            UK
          </button>
          <p
            class="text-white"
          />
           
          <svg
            class="arrow-right"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M504 256C504 119 393 8 256 8S8 119 8 256s111 248 248 248 248-111 248-248zm-448 0c0-110.5 89.5-200 200-200s200 89.5 200 200-89.5 200-200 200S56 366.5 56 256zm72 20v-40c0-6.6 5.4-12 12-12h116v-67c0-10.7 12.9-16 20.5-8.5l99 99c4.7 4.7 4.7 12.3 0 17l-99 99c-7.6 7.6-20.5 2.2-20.5-8.5v-67H140c-6.6 0-12-5.4-12-12z"
            />
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class=" col-6"
      style="transform: none;"
    >
      <div
        class="countries"
      >
        <button
          class="navigate"
          type="button"
        >
          UK
        </button>
        <p
          class="text-white"
        />
         
        <svg
          class="arrow-right"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M504 256C504 119 393 8 256 8S8 119 8 256s111 248 248 248 248-111 248-248zm-448 0c0-110.5 89.5-200 200-200s200 89.5 200 200-89.5 200-200 200S56 366.5 56 256zm72 20v-40c0-6.6 5.4-12 12-12h116v-67c0-10.7 12.9-16 20.5-8.5l99 99c4.7 4.7 4.7 12.3 0 17l-99 99c-7.6 7.6-20.5 2.2-20.5-8.5v-67H140c-6.6 0-12-5.4-12-12z"
          />
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
